### PR TITLE
Add dilation for conv_trans_op

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/6189
fix https://github.com/PaddlePaddle/Paddle/issues/6423

### CODE REVIEW

**Comment by jacquesqiao** [comment]
It seems no code is using attribute dilations?

**Comment by chengduoZH** [comment]
The code is not fully implemented yet. @jacquesqiao

File: python/paddle/v2/fluid/tests/test_conv2d_transpose_op.py [hunk 1]
```diff
@@ -104,11 +108,60 @@ def init_test_case(self):
...
+        self.dilations = [1, 1]
+        self.input_size = [2, 3, 5, 5]  # NCHW
+        f_c = self.input_size[1]
+        self.filter_size = [f_c, 6, 3, 3]
+
+    def init_op_type(self):
+        self.op_type = "conv2d_transpose_cudnn"
+
+
+# #cudnn v5 does not support dilation conv.
```
Comment (wanghaoshuang) [inline python/paddle/v2/fluid/tests/test_conv2d_transpose_op.py]: 直接注释不太合适吧？对不同CUDNN_VERSION做测试，你可以参考这里：
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/gserver/tests/test_LayerGrad.cpp#L198

Comment (chengduoZH) [reply python/paddle/v2/fluid/tests/test_conv2d_transpose_op.py]: 我觉得可以在下一个PR里面添加一个从python端获取CUDNN_VERSION的接口

**Pull request merged**